File: app/src/main/java/com/rendersoncs/reportform/constants/ReportConstants.java
package com.rendersoncs.reportform.constants;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.database.DatabaseReference;
import com.rendersoncs.reportform.login.util.LibraryClass;
import com.rendersoncs.reportform.login.util.User;

public class ReportConstants {

    public static class ConstantsProvider{
        public static final String PACKAGE_FILE_PROVIDER = "com.rendersoncs.reportform.FileProvider";
    }

    public static class ConstantsFireBase{
        public static final String URL = "https://reportform-20b2a.firebaseio.com/users";
        public static final String JSON_FIRE = "reportApp";
        public static final String FIRE_DESCRIPTION = "description";
        public static final String FIRE_TITLE = "title";
    }

    public static class ConstantsBundle {
        public static final String REPORT_ID = "reportId";
    }

    public static class ConstantsCharacters{
        public static final int LIMITS_TEXT = 14;
    }

    public static class LIST_ITEMS {
        public static final String TITLE = "title";
        public static final String DESCRIPTION = "description";
        public static final String CONFORMITY = "conformity";
        public static final String NOTE = "note";
        public static final String PHOTO = "photo";
    }

    public  static  class TEST {
        FirebaseAuth mAuth = FirebaseAuth.getInstance();
        FirebaseUser user = mAuth.getCurrentUser();
        String userId = user.getUid();
        String test = ReportConstants.ConstantsFireBase.URL + "/" + userId;

        public final String DB = test;

    }
}

File: app/src/main/java/com/rendersoncs/reportform/view/services/constants/ReportConstants.kt
package com.rendersoncs.reportform.view.services.constants

class ReportConstants {
    object PACKAGE {
        const val FILE_PROVIDER = "com.rendersoncs.reportform.FileProvider"
    }

    object FIREBASE {
        const val URL = "https://reportform-20b2a.firebaseio.com/users"
        const val FIRE_USERS = "users"
        const val FIRE_CREDENTIAL = "credential"
        const val FIRE_PHOTO = "photoUrl"
        const val FIRE_LIST = "list"
        const val FIRE_NAME = "name"
    }

    object REPORT {
        const val REPORT_ID = "reportId"
    }

    object CHARACTERS {
        const val LIMITS_TEXT = 14
    }

    object ITEM {
        const val POSITION = "position"
        const val COMPANY = "company"
        const val EMAIL = "email"
        const val CONTROLLER = "controller"
        const val DATE = "date"
        const val TITLE = "title"
        const val DESCRIPTION = "description"
        const val KEY = "key"
        const val CONFORMITY = "conformity"
        const val NOTE = "note"
        const val PHOTO = "photo"
        const val OPT_NUM1 = 1
        const val OPT_NUM2 = 2
        const val OPT_NUM3 = 3
    }

    object PHOTO {
        const val REQUEST_CODE_GALLERY = 2006
        const val REQUEST_CODE_CAMERA = 2000
        const val REQUEST_PERMISSIONS_CAMERA = 0
        const val NOT_PHOTO = "notPhoto"
    }
}